server/policies/cancan.ts

```diff
@@ -74,21 +78,58 @@ export class CanCan {
     target: Model | null | undefined,
     options = {}
   ) => {
-    const matchingAbilities = this.abilities.filter(
-      (ability) =>
-        performer instanceof ability.model &&
-        (ability.target === "all" ||
-          target === ability.target ||
-          target instanceof (ability.target as any)) &&
-        (ability.action === "manage" || action === ability.action)
+    const matchingAbilities = this.getMatchingAbilities(
+      performer,
+      action,
+      target
     );
 
     // Check conditions only for matching abilities
-    return matchingAbilities.some(
-      (ability) =>
-        !ability.condition ||
-        ability.condition(performer, target, options || {})
+    const conditions = uniq(
```

Comment:
All of these changes are to allow policies to return `membershipIds` instead of just booleans